src/rgw/rgw_formats.cc

```diff
@@ -128,7 +128,7 @@ void RGWFormatter_Plain::dump_float(std::string_view name, double d)
 
 void RGWFormatter_Plain::dump_string(std::string_view name, std::string_view s)
 {
-  dump_format(name, "%s", s.data());
+  dump_format(name, "%.*s", s.size(), s.data());
```

Comment:
are you sure it is ".*" and not "*"?

Reply:
looking at http://www.cplusplus.com/reference/cstdio/printf/

the distinction is between the `.precision` format and the `width` format. width talks about a _minimum_ length of characters to print. precision has a special case for strings:
> For s: this is the maximum number of characters to be printed. By default all characters are printed until the ending null character is encountered.